contracts/fungible-token/io/src/lib.rs

```diff
@@ -34,52 +58,102 @@ pub enum FTAction {
     },
     Burn {
         amount: u128,
-        from: ActorId,
     },
     Transfer {
+        tx_id: Option<TxId>,
         from: ActorId,
         to: ActorId,
         amount: u128,
     },
     Approve {
+        tx_id: Option<TxId>,
         to: ActorId,
         amount: u128,
     },
-    TotalSupply,
     BalanceOf(ActorId),
     AddAdmin {
-        admin: ActorId,
-    }
+        admin_id: ActorId,
+    },
+    DeleteAdmin {
+        admin_id: ActorId,
+    },
 }
 
 #[derive(Debug, Encode, Decode, TypeInfo)]
 #[codec(crate = gstd::codec)]
 #[scale_info(crate = gstd::scale_info)]
-pub enum FTEvent {
-    Transfer {
+pub enum FTReply {
+    Burned {
+        amount:u128
+    },
+    Transferred {
         from: ActorId,
         to: ActorId,
         amount: u128,
     },
-    Approve {
+    Approved {
         from: ActorId,
         to: ActorId,
         amount: u128,
     },
-    TotalSupply(u128),
+    AdminAdded {
+        admin_id: ActorId,
+    },
+    AdminRemoved {
+        admin_id: ActorId,
+    },
     Balance(u128),
-    AdminAdded
 }
 
-#[derive(Debug, Clone, Default, Encode, Decode, TypeInfo)]
+#[derive(Debug, Encode, Decode, TypeInfo, MaxEncodedLen, Clone)]
 #[codec(crate = gstd::codec)]
 #[scale_info(crate = gstd::scale_info)]
-pub struct IoFungibleToken {
-    pub name: String,
-    pub symbol: String,
-    pub admins: Vec<ActorId>,
-    pub total_supply: u128,
-    pub balances: Vec<(ActorId, u128)>,
-    pub allowances: Vec<(ActorId, Vec<(ActorId, u128)>)>,
-    pub decimals: u8,
+pub enum FTError {
+    NotAdmin,
+    NotEnoughBalance,
+    ZeroAddress,
+    NotAllowedToTransfer,
+    AdminAlreadyExists,
+    CantDeleteYourself,
+    TxAlreadyExists,
+}
+
+#[derive(Encode, Decode, TypeInfo)]
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
+pub enum Query {
+    Name,
+    Symbol,
```

Comment:
I think it would be useful to add one more field here to get the description
```suggestion
    Symbol,
    Description,
```